### ISSUE TITLE
\lr show vimtex reverse search failed (is MuPDF open?) 

### ISSUE THREAD

**Issue by hanqing0521** [lervag/vimtex#415]
### Explain the issue

I install mupdf and changing texviewer to mupdf. If i run `\lr` show `vimtex reverse search failed (is MuPDF open?)`  but mupdf can show me the pdf.

I also find that  if i close the mupdf, then i run `\ll` the viewer can not work again.

1. changing viewer to mupdf and `\ll` (or `:VimtexCompile`). Then, mupdf show pdf. run `\lr`

2. normal operation you know, it show just work as your design.

3. it can not work  show me `vimtex reverse search failed (is MuPDF open?)` _And_  after close the mupdf window, run `\ll` to compile agian it show me `already running for 'hello_world.tex'`  mupdf will not work agian.  but '\\lv` (`:VimtexView`) can work.

### Minimal working example

Please provide a minimal working LaTeX example, e.g.

```tex
\documentclass{article}
\usepackage{graphics}
\begin{document}
Hello world!   Fig.\ref{Eq:cos}
    \begin{equation}
        f(x)=cos(x)
        \label{Eq:cos}
    \end{equation}
\end{document}
```

### Minimal vimrc file

```vim
set nocompatible
let &rtp  = '~/.vim/bundle/vimtex,' . &rtp
let &rtp .= ',~/.vim/bundle/vimtex/after'
filetype plugin indent on
syntax enable
let g:vimtex_view_method = 'mupdf'
```

**Comment by lervag** [comment]
First: I've updated your formatting. In particular, I've removed the numbers from the minimal vimrc file. Please be a little bit more careful when you submit issues that your formatting is done properly. I like to think that if people don't spend time properly formulating and formatting issues and questions for me, why should I spend my time responding? Note: It is MUCH easier to read a properly formatted issue!

Could you do the following (with your provided minimal setup, of course):

0. Ensure the pdf file is available (compile the document once)

1. Open tex file in Vim

2. Do `\lv` (to open mupdf viewer)

3. Do `\li` to show vimtex info, please print it here

Note in particular that the output from `\li` should show an entry similar to `xwin_id` which should not be zero. If the number is nonzero, then continue:

4. Close mupdf window

5. Repeat steps 2 and 3

**Comment by hanqing0521** [comment]
I apologize for my carelessness. 

### operations

1. finished

2. ok

3. mupdf work

4. \li results
```
b:vimtex : hello_world
  pid : 0
  root : '/home/lixun/Documents/tex/test'
  aux : '/home/lixun/Documents/tex/test/hello_world.aux'
  log : '/home/lixun/Documents/tex/test/hello_world.log'
  out : '/home/lixun/Documents/tex/test/hello_world.pdf'
  tex : '/home/lixun/Documents/tex/test/hello_world.tex'
  base : 'hello_world.tex'
  viewer
    page : ''
    xwin_id : '150995203'
    class : 'MuPDF'
    cmd_start : 'mupdf  ''/home/lixun/Documents/tex/test/hello_world.pdf'''
    cmd_synctex_view : 'synctex view -i 2:2:''/home/lixun/Documents/tex/test/hello_world.tex'' -o ''/home/lixun/Docu
ments/tex/test/hello_world.pdf'' | grep -m1 ''Page:'' | sed ''s/Page://'' | tr -d ''
'''
    init : function('20')
    view : function('21')
    start : function('22')
    forward_search : function('23')
    reverse_search : function('24')
    latexmk_callback : function('25')
    latexmk_append_argument : function('26')
    focus_viewer : function('<SNR>37_focus_viewer')
    focus_vim : function('<SNR>37_focus_vim')
    xwin_exists : function('<SNR>37_xwin_exists')
    xwin_get_id : function('<SNR>37_xwin_get_id')
```
**xwin_id is not zero**
5.  **I do not understand your meaning that should I close vim and open agian or just do \lv agian**
if closing vim after closeing mupdf window, then open new vim, of course it is normal.  
**just close mupdf and \lv again**
mupdf work with same xwin_id ( it should be new i think)
```
b:vimtex : hello_world
  pid : 0
  root : '/home/lixun/Documents/tex/test'
  aux : '/home/lixun/Documents/tex/test/hello_world.aux'
  log : '/home/lixun/Documents/tex/test/hello_world.log'
  out : '/home/lixun/Documents/tex/test/hello_world.pdf'
  tex : '/home/lixun/Documents/tex/test/hello_world.tex'
  base : 'hello_world.tex'
  viewer
    page : ''
    xwin_id : '150995203'
    class : 'MuPDF'
    cmd_start : 'mupdf  ''/home/lixun/Documents/tex/test/hello_world.pdf'''
    cmd_synctex_view : 'synctex view -i 2:2:''/home/lixun/Documents/tex/test/hello_world.tex'' -o ''/home/lixun/Docu
ments/tex/test/hello_world.pdf'' | grep -m1 ''Page:'' | sed ''s/Page://'' | tr -d ''
'''
    init : function('20')
    view : function('21')
    start : function('22')
    forward_search : function('23')
    reverse_search : function('24')
    latexmk_callback : function('25')
    latexmk_append_argument : function('26')
    focus_viewer : function('<SNR>37_focus_viewer')
    focus_vim : function('<SNR>37_focus_vim')
    xwin_exists : function('<SNR>37_xwin_exists')
    xwin_get_id : function('<SNR>37_xwin_get_id')
    xwin_send_keys : function('<SNR>37_xwin_send_keys')

请按 ENTER 或其它命令继续
```

### Other things
Though I close the mupdf window, doing \li it show  me the killed xwin_id

**Comment by lervag** [comment]
Thanks. I did not mean or write that you should close Vim after step 4, but close the mupdf window. I wanted to see if closing it and reopening with `\lv` would create a new xwin_id. It seems it did not.

Which version of xdotool are you using?

New test:

1. Open tex file
2. Open mupdf with `\lv`
3. Close mupdf window
4. Start compilation with `\ll`

Does the mupdf window open?

Which OS are you on, and which version of Vim?

**Comment by hanqing0521** [comment]
# <h1> xdotool version
```
  manual xdotool --version
xdotool version 3.20151124.1
```
# <h1> my vim and os
```
 manual vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Aug 20 2015 09:52:41)
包含补丁: 1-207, 209-801, 803-808, 810-827
修改者 <bugzilla@redhat.com>
编译者 <bugzilla@redhat.com>
巨型版本 无图形界面。  可使用(+)与不可使用(-)的功能:
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
-balloon_eval    +float           +mouse_urxvt     -tag_any_white
-browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         +gettext         -mzscheme        +textobjects
-clientserver    -hangul_input    +netbeans_intg   +title
-clipboard       +iconv           +path_extra      -toolbar
+cmdline_compl   +insert_expand   +perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python/dyn      +viminfo
+cscope          +lispindent      +python3/dyn     +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      +lua/dyn         +rightleft       +windows
+diff            +menu            +ruby/dyn        +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      -sniff           -xsmp
+eval            +mouse_dec       +startuptime     -xterm_clipboard
+ex_extra        +mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    -xpm
     系统 vimrc 文件: "/etc/vimrc"
     用户 vimrc 文件: "$HOME/.vimrc"
 第二用户 vimrc 文件: "~/.vim/vimrc"
      用户 exrc 文件: "$HOME/.exrc"
         $VIM 预设值: "/etc"
  $VIMRUNTIME 预设值: "/usr/share/vim/vim74"
编译方式: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -O2 -g -pipe -Wall -Werror=format-security -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
链接方式: gcc   -L. -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,--enable-new-dtags -Wl,-z,relro   -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -L/usr/local/lib -Wl,--as-needed -o vim        -lm -lnsl  -lselinux   -lncurses -lacl -lattr -lgpm -ldl   -Wl,--enable-new-dtags -Wl,-z,relro  -Wl,-z,relro  -fstack-protector-strong -L/usr/local/lib  -L/usr/lib64/perl5/CORE -lperl -lpthread -lresolv -lnsl -ldl -lm -lcrypt -lutil -lc        

```
*OS* 
```
  manual uname -a
Linux lixun-linux-laptop 4.4.2-301.fc23.x86_64 #1 SMP Tue Feb 23 19:00:38 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```
# <h1> New test
1. ^_^
2. ^_^
3. -_-
4. \ll(:VimtexCompile) work and show me the pdf by mupdf.

# <h1> Just my findging
if running the \ll and close the mupdf, \ll agian program tell me `latexmk compile: already running for `hello_world.tex' ` 
*I mean may the old compile process is not killed or restarted* if i use my okular view pdf, i can see run \ll again , vimtex will killed old process and run a new process or restart. But now, I should kill process by myself.

**Comment by lervag** [comment]
I'm a little bit confused. What do you mean "I mean may the old compile process is not killed or restarted if i use my okular view pdf"?

if you do `\ll`, then repeat `\ll`, vimtex should first start compiling, then stop compiling. You should not need to stop the process yourself. If this does not work, please open a new issue and explain how to reproduce, including the minimal example.

However, the issue as I understood it has to do with the reverse search with `\lr`. This should work, and it works for me. I am not able to reproduce your problem.

**Comment by hanqing0521** [comment]
I just tell you if i use mupdf view my pdf, then I repeat \ll it just show me compile processing runing, do not stop.  *and i think there are some relation between the two problems.

i do not konw why  \lr can not work always tell me  `vimtex reverse search failed (is MuPDF open?) `
my another pc asole meets that problem.

**Comment by lervag** [comment]
Ok. I don't think that the problem with mupdf is related to the problem with repeating `\ll` for toggling compilation. I'm pretty sure about that, actually. Thus I would be happy if you could open a new issue about the compilation problem.

Back to the mupdf problem: Do I understand correctly that the only problem you have is with `\lr`? That is, both forward search and opening with `\lv` works as expected?

**Comment by hanqing0521** [comment]
I understand it. it will be a new issue.

\lv is ok. but \lr have problem

# <h1>there are some new info that may help you find where should be changed

1. open vim  
`vim -u minivimrc hello_world.tex `
2. \lv 
3. \lr
it will open a new buffer 
```
:buffers
  1 %a   "hello_world.tex"              第 1 行
  2 #    "SyncTeX ERROR: No SyncTeX available for /home/lixun/Documents/tex/test/hello_world.pdf^@usage: synctex <subcommand> [options] [args]^@Synchronize TeXnology comman
d-line client, version 1.15^@^@The Synchronization TeXnology by Jérôme Laurens is a new feature of recent TeX engines.^@It allows to synchronize between input and output, w
hich means to^@navigate from the source document to the typeset material and vice versa.^@^@synctex edit: backwards or reverse synchronization,^@command sent by the viewer
to edit the source corresponding to the position under the mouse^@^@^@usage: synctex edit -o page:x:y:file [-d directory] [-x editor-command] [-h offset:context]^@^@-o page
:x:y:file^@       specify the page and coordinates of the point under the mouse.^@       page is 1 based.^@       Coordinates x and y are counted from the top left corner o
f the page.^@       Their unit is the big point (72 dpi).^@       ^@    
```

**Comment by lervag** [comment]
Ah, I see! Can you add a `.latexmkrc` file to your home directory with the following content:

```perl
$pdflatex = 'pdflatex -synctex=1 -interaction=nonstopmode --shell-escape %O %S';
```

I think that should solve your issue(s).

**Comment by lervag** [comment]
I think perhaps this might solve both of the problems you had. `-synctex=1` enabled synctex, which is important for the `\lr` to work. The `-interaction=nonstopmode` might also be important, since without it, any errors in your document will make the compilation hang.

**Comment by hanqing0521** [comment]
i am sorry to tell you the bad news that it did not work well
Of course it improve a little

The problem is 

after i run \lv, run \lr  everything is good
*but*  firstly, \ll also have problem. \ll repeat \ll do not kill process. and then after \ll \lr show no mupdf window open.

**Comment by lervag** [comment]
Ok, as I just mentioned in #431, you no longer need the beforementioned `.latexmkrc` file. I've added the mentioned flags to vimtex directly.

So, it seems to me that we have solved the original issue, that is, that `\lr` did not work. This is the issue that is mentioned in the title. I propose that you open a _new_ issue for the remaining problem. Please explain in detail how to repeat your problem.
